internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -154,17 +166,28 @@ func (r *PreprovisioningImageReconciler) getIronicIgnitionConfig(log logrus.Fiel
 	return string(config), err
 }
 
-func (r *PreprovisioningImageReconciler) setImage(generation int64, status *metal3_v1alpha1.PreprovisioningImageStatus, infraEnv aiv1beta1.InfraEnv) error {
+func (r *PreprovisioningImageReconciler) setImage(image *metal3_v1alpha1.PreprovisioningImage, infraEnv aiv1beta1.InfraEnv) error {
 	r.Log.Infof("Updating PreprovisioningImage ImageUrl to: %s", infraEnv.Status.ISODownloadURL)
-	status.ImageUrl = infraEnv.Status.ISODownloadURL
-	status.Architecture = infraEnv.Spec.CpuArchitecture
-	status.Format = metal3_v1alpha1.ImageFormatISO
+	image.Status.Architecture = infraEnv.Spec.CpuArchitecture
+	if funk.Contains(image.Spec.AcceptFormats, metal3_v1alpha1.ImageFormatISO) {
```

Comment:
Hmm, pretty sure image.Spec.AcceptFormats can contain both ISO and InitRD.
Since the ImageUrl is required for both formats (different artifacts) We should decide which one we set in case it accepts both.

Reply:
I consciously did it this way to prefer ISO. Is there a significant difference between the two if both are accepted?

Reply:
Good call